core/src/serde_support.rs

```diff
@@ -0,0 +1,589 @@
+        let bytes: [u8; Self::SIZE] = decoded.try_into().map_err(|_| {
+            SerdeError::invalid_length(decoded_len, &byte_length_str.as_str())
+        })?;
+        ViewKey::from_bytes(&bytes)
+            .map_err(|err| SerdeError::custom(format!("{err:?}")))
+    }
+}
+
+impl Serialize for NoteType {
+    fn serialize<S: Serializer>(
+        &self,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        match self {
+            Self::Transparent => {
+                serializer.serialize_unit_variant("NoteType", 0, "Transparent")
+            }
+            Self::Obfuscated => {
+                serializer.serialize_unit_variant("NoteType", 1, "Obfuscated")
+            }
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for NoteType {
+    fn deserialize<D: Deserializer<'de>>(
+        deserializer: D,
+    ) -> Result<Self, D::Error> {
+        let value = String::deserialize(deserializer)?;
+        match value.as_str() {
+            "Obfuscated" => Ok(NoteType::Obfuscated),
+            "Transparent" => Ok(NoteType::Transparent),
+            v => Err(SerdeError::unknown_variant(
+                v,
+                &["Transparent", "Obfuscated"],
+            )),
+        }
+    }
+}
+
+impl Serialize for StealthAddress {
+    fn serialize<S: Serializer>(
+        &self,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        let s = bs58::encode(self.to_bytes()).into_string();
+        serializer.serialize_str(&s)
+    }
+}
+
+impl<'de> Deserialize<'de> for StealthAddress {
+    fn deserialize<D: Deserializer<'de>>(
+        deserializer: D,
+    ) -> Result<Self, D::Error> {
+        let s = String::deserialize(deserializer)?;
+        let decoded =
+            bs58::decode(&s).into_vec().map_err(SerdeError::custom)?;
+        let decoded_len = decoded.len();
+        let byte_length_str = Self::SIZE.to_string();
+        let bytes: [u8; Self::SIZE] = decoded.try_into().map_err(|_| {
+            SerdeError::invalid_length(decoded_len, &byte_length_str.as_str())
+        })?;
+        StealthAddress::from_bytes(&bytes)
+            .map_err(|err| SerdeError::custom(format!("{err:?}")))
+    }
+}
+
+// To serialize and deserialize u64s as big ints:
+// https://github.com/dusk-network/rusk/issues/2773#issuecomment-2519791322.
+struct Bigint(u64);
+
+impl Serialize for Bigint {
+    fn serialize<S: Serializer>(
+        &self,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        let s: String = format!("{}n", self.0);
+        s.serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for Bigint {
+    fn deserialize<D: Deserializer<'de>>(
+        deserializer: D,
+    ) -> Result<Self, D::Error> {
+        let mut s = String::deserialize(deserializer)?;
+        let last_char = s.pop().ok_or_else(|| {
+            SerdeError::invalid_value(
+                Unexpected::Str(&s),
+                &"a bigint ending with character 'n'",
```

Comment:
I think here the error should say that we are expecting the string not to be empty.